### DISCUSSION TITLE
Skip creating the database

### DISCUSSION THREAD

**Discussion by yadavsanket** [oracle/docker-images#2173] [Q&A]
I am trying to skip creating the database when we launch `docker run ....` command. I tried commenting out the statements that create the database - basically the statements starting from below line -
`if [ -f $ORACLE_BASE/oradata/.${ORACLE_SID}${CHECKPOINT_FILE_EXTN} ]; then`
up to the  matching `fi;` statement in runOracle.sh.

However, there are a lot of checks which are thwarting my attempts to do so. I want to do this as I already have my own database creation scripts. And I do not want to unnecessarily keep two databases in the container.

**Is there an elegant way to skip creating the default database and use my custom DB creation scrips?** 

Side note: I tried specifying my custom scripts by mapping to /opt/oracle/scripts/setup in the hopes of automatically creating my custom DB along with default DB but even that attempt failed.

**Comment by gvenzl** [comment]
Hi @yadavsanket,

The most elegant way of creating your database is to extend the existing image and overwrite the `CMD` instruction with your script.
Alternatively, you could also extend the image and overwrite the `$RUN_FILE` (`/opt/oracle/runOracle.sh`) with your own.
Last but not least, you could create your database once, keep that in a volume and start a container pointing to that image.

**Comment by gyurovb** [comment]
`docker run -d -it --name <the name here> <your image here> /bin/bash`
